frontend/src/hooks/useTypingEffect.test.ts

```diff
@@ -0,0 +1,150 @@
+    });
+
+    expect(result.current).toBe(message.slice(0, -1));
+
+    act(() => {
+      jest.advanceTimersByTime(1); // include the last character
+    });
+
+    expect(result.current).toBe(message);
+  });
+
+  it("should loop through strings when multiple are provided", () => {
+    const messages = ["Hello", "World"];
+
+    const { result } = renderHook(() => useTypingEffect(messages));
+
+    const msToRunFirstString = messages[0].length * 100 * 0.1;
+
+    // Fast-forward to end of first string
+    act(() => {
+      jest.advanceTimersByTime(msToRunFirstString);
+    });
+
+    expect(result.current).toBe(messages[0]); // Hello
+
+    // Fast-forward through the delay and through the second string
+    act(() => {
+      // TODO: Improve to clarify the expected timing
+      jest.runAllTimers();
+    });
+
+    expect(result.current).toBe(messages[1]); // World
+  });
+
+  it("should call setTypingActive with false when typing completes without loop", () => {
+    const setTypingActiveMock = jest.fn();
+
+    renderHook(() =>
+      useTypingEffect(["Hello, world!", "This is a test message."], {
+        loop: false,
+        setTypingActive: setTypingActiveMock,
+      }),
+    );
+
+    expect(setTypingActiveMock).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(setTypingActiveMock).toHaveBeenCalledWith(false);
+    expect(setTypingActiveMock).toHaveBeenCalledTimes(1);
+  });
+
+  it("should call addAssistantMessageToChat with the typeThis argument when typing completes without loop", () => {
+    const addAssistantMessageToChatMock = jest.fn();
+
+    renderHook(() =>
+      useTypingEffect(["Hello, world!", "This is a test message."], {
+        loop: false,
+        // Note that only "Hello, world!" is typed out (the first string in the array)
+        typeThis: { content: "Hello, world!", sender: "assistant" },
+        addAssistantMessageToChat: addAssistantMessageToChatMock,
+      }),
+    );
+
+    expect(addAssistantMessageToChatMock).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(addAssistantMessageToChatMock).toHaveBeenCalledTimes(1);
+    expect(addAssistantMessageToChatMock).toHaveBeenCalledWith({
+      content: "Hello, world!",
+      sender: "assistant",
+    });
+  });
+
+  it("should call takeOneAndType when typing completes without loop", () => {
+    const takeOneAndTypeMock = jest.fn();
+
+    renderHook(() =>
+      useTypingEffect(["Hello, world!", "This is a test message."], {
+        loop: false,
+        takeOneAndType: takeOneAndTypeMock,
+      }),
+    );
+
+    expect(takeOneAndTypeMock).not.toHaveBeenCalled();
+
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    expect(takeOneAndTypeMock).toHaveBeenCalledTimes(1);
+  });
+
+  // Implementation is not clear on how to handle this case
+  it.todo("should handle typing with loop");
```

Comment:
It is not clear what the purpose of the `loop` argument is. The only instance of the hook we use is in `src/components/ChatInterface.tsx`, and loop is explicitly set to `false`.